earth_enterprise/src/common/SharedString.h

```diff
@@ -37,35 +38,77 @@
 // much faster as we only need to compare the identifiers.
 class SharedString {
 protected:
-   class StringStorage {
+    class stringStorageRWLock
+    {
+      private:
+        pthread_rwlock_t rwlock;
```

Comment:
I thought Justin already wrote this but I am not seeing it. Might be easier to do this with boost. Wouldn't be a bad idea to add it to khThread also

Reply:
I had previously tried to implement boost's read/write locks, but quite a few linker issues regarding the unit tests occurred when adding that dependency to SharedString. Perhaps moving this to khThread may avoid that.